src/goto-instrument/contracts/dynamic-frames/dfcc_root_object.h

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking
+
+Author: Remi Delmas, delmasrd@amazon.com
+
+Date: March 2023
+
+\*******************************************************************/
+
+/// \file
+/// Utility functions that compute root object expressions for assigns clause
+/// targets and LHS expressions.
+
+#ifndef CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_ROOT_OBJECT_H
+#define CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_ROOT_OBJECT_H
+
+#include <util/irep.h>
+
+#include <unordered_set>
+
+class exprt;
+class namespacet;
+class messaget;
+
+/// Computes a set of root object expressions for an assignment left hand side
+/// expression.
+std::unordered_set<exprt, irep_hash> dfcc_assignment_lhs_root_objects(
+  const exprt &lhs,
+  const namespacet &ns,
+  messaget &log);
```

Comment:
We typically pass around a `message_handlert`, not a `messaget`. (The latter could have an incomplete message in flight.) Also, do you ever actually use this?

Reply:
These were leftovers from debugging. removed.